src/components/buttons/browse-the-book.cjsx

```diff
@@ -2,27 +2,27 @@ React = require 'react'
 BS = require 'react-bootstrap'
 
 module.exports = React.createClass
-  displayName: 'BrowseTheBookButton'
+  displayName: 'BrowseTheBook'
 
   contextTypes:
     router: React.PropTypes.func
 
   propTypes:
-    courseId: React.PropTypes.string.isRequired
-    page:     React.PropTypes.string
+    courseId:  React.PropTypes.string
+    chapterId: React.PropTypes.number
+    sectionId: React.PropTypes.number
+    page:      React.PropTypes.string
+    unstyled:  React.PropTypes.bool
 
   render: ->
-    link = if @props.page
-      @context.router.makeHref('viewReferenceBookPage',
-        {courseId: @props.courseId, cnxId:@props.page})
+    courseId = @props.courseId or @context.router.getCurrentParams().courseId
+    # the router is smart enough to figure out which props are present and return the best route
+    linkType = if @props.page then 'viewReferenceBookPage' else
+      if @props.section then 'viewReferenceBookSection' else 'viewReferenceBook'
+    link = @context.router.makeHref( linkType, courseId: courseId, cnxId: @props.page, section:@props.section )
+    linkProps = {target:'_blank', className:'view-reference-guide', href: link}
+    text = @props.children or 'Browse the Book'
+    if @props.unstyled
+      <a {...linkProps}>{text}</a>
     else
-      @context.router.makeHref('viewReferenceBook',
-        {courseId: @props.courseId})
-    <BS.Button
-      bsStyle='primary'
-      target="_blank"
-      className='view-reference-guide'
-      href={link}
-    >
-      Browse the Book
-    </BS.Button>
+      <BS.Button bsStyle='primary', {...linkProps}>{text}</BS.Button>
```

Comment:
The comma was an oversight, I've removed it.  

The react-bootstrap components translate the `bsStyle='primary'` prop into `btn-primary`, and the `btn` class is applied since it's a `Button` component.  If you think it'd be clearer I can use explicit classes though.